src/MOI_wrapper.jl

```diff
@@ -0,0 +1,229 @@
+"""
+Constructs a Differentiable Optimizer model from a MOI Optimizer.
+Supports `forward` and `backward` methods for solving and differentiating the model respectectively.
+
+## Note 
+Currently supports differentiating linear and quadratic programs only.
+"""
+
+const VI = MOI.VariableIndex
+const CI = MOI.ConstraintIndex
+
+
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    optimizer::MOI.Bridges.LazyBridgeOptimizer
+    primal_optimal::Array{Float64}  # solution
+    dual_optimal::Array{Float64}  
+    var_idx::Vector{VI}
+    con_idx::Vector{CI}
+
+    function Optimizer(optimizer_constructor::Type)
+        new(
+            MOI.instantiate(optimizer_constructor, with_bridge_type=Float64),
+            zeros(0),
+            zeros(0),
+            Vector{VI}(),
+            Vector{CI}()
+        )
+    end
+end
+
+
+function MOI.add_variable(model::Optimizer)
+    vi = MOI.add_variable(model.optimizer)
+    push!(model.var_idx, vi)
+    return vi
+end
+
+
+function MOI.add_variables(model::Optimizer, N::Int)
+    return VI[MOI.add_variable(model) for i in 1:N]
+end
+
+
+function MOI.add_constraint(model::Optimizer, f::F, s::S) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
```

Comment:
You can just do `f::MOI.AbstractFunction` as you don't need `F`

Reply:
fixed